R/serialization.R

```diff
@@ -0,0 +1,41 @@
+# This file will be autogenerated.
+
+#' Serialize a model to the given filename.
+#'
+#' @title Serialize.
+#' @param filename Input filename.
+#' @param model_in Input model.
+Serialize <- function(filename, model_in) {
+  if (attributes(model_in)$type == "GaussianKernel") {
+    con <- file(as.character(filename), "wb")
+    serialize(SerializeGaussianKernelPtr(model_in), con)
+    close(con)
+  }
+  else if (attributes(model_in)$type == "LARS") {
+    con <- file(as.character(filename), "wb")
+    serialize(SerializeLARSPtr(model_in), con)
+    close(con)
+  }
+}
+
+
+#' Unserialize a model to the given filename.
+#'
+#' @title Unserialize.
+#' @param filename Input filename.
+#' @return model_ptr Output model.
+Unserialize <- function(filename) {
+  con <- file(as.character(filename), "rb")
+  model <- unserialize(con)
+
+  if (attributes(model)$type == "GaussianKernel") {
```

Comment:
Can you try to implement a similar switch statement as above?

Reply:
Updated this part.